app/controllers/oauth2_controller.rb

```diff
@@ -35,8 +44,11 @@ def destroy
 
   private
 
-    def authenticate_with_credentials(identification, password)
-      login = Login.where(identification: identification).first || LoginNotFound.new
+    def authenticate_with_credentials(identification, password, provider, client)
+      login = Login.where(provider: provider, identification: identification, client: client).first
```

Comment:
where().first can always be cleaned up to `Login.find_by(provider: provider, identification: identification, client: client)`.

Man, ES6 has me all thinking you could do something like `Login.find_by({provider, identification, client})`... but you can't :/